warehouse/age-off-utils/src/main/java/datawave/age/off/util/AgeOffFileGenerator.java

```diff
@@ -0,0 +1,69 @@
+package datawave.age.off.util;
+
+import java.io.IOException;
+import java.io.Writer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Using an AgeOffFileConfiguration, writes an xml age off file containing an optional parent reference and a set of ordered rules.
+ */
+public class AgeOffFileGenerator {
+    private static final Logger log = LoggerFactory.getLogger(AgeOffFileGenerator.class);
+
+    private final AgeOffFileConfiguration configuration;
+    private final String indent;
+    private final Writer writer;
+
+    public AgeOffFileGenerator(AgeOffFileConfiguration configuration) {
+        this.configuration = configuration;
+        this.indent = this.configuration.getIndentation();
+        this.writer = this.configuration.getWriter();
+    }
+
+    /**
+     * Writes the file to the writer.
+     *
+     * @throws IOException
+     *             i/o exception with writer
+     */
+    public void format() throws IOException {
+        openConfigurationElement();
```

Comment:
I like this style of declarative generation of the XML. If we wanted to one day take this further, we could make an object hierarchy that reflects the structure of the XML file and then just let something like jaxb or jackson serialize it out as XML at the end. Then adding a rule or a parent would just be manipulating Java objects rather than needing to properly format each line of output. 

I do think this is a good first step though if we don't want to do the whole object style initially - it's probably a significant rework.

Reply:
The object hierarchy is tough because there are so many different implementations of filters that make the rule so different.

I tried changing to XMLStreamWriter but it didn't support new lines nor indentation very well.  I reverted those changes for now.  

This is being left unresolved for now while I try a few alternatives.

Reply:
The implementation in the latest commit, using xml Document objects and a Transformer, worked out better than XMLStreamWriter.